frontend/src/static/js/components/webstatus-stats-page.ts

```diff
@@ -149,18 +171,46 @@ export class StatsPage extends LitElement {
       )) {
         // Append the new data to existing data
         const existingData =
-          this.globalFeatureSupport.get(globalFeatureSupportKey(browser)) || [];
-        this.globalFeatureSupport.set(globalFeatureSupportKey(browser), [
+          this.globalFeatureSupport.get(statsDataKey(browser)) || [];
+        this.globalFeatureSupport.set(statsDataKey(browser), [
           ...existingData,
           ...page,
         ]);
       }
-      this.globalFeatureSupportChartDataObj =
-        this.createGlobalFeatureSupportDataFromMap();
+      this.globalFeatureSupportChartDataObj = this.createDisplayDataFromMap(
+        this.globalFeatureSupport,
+        true,
+      );
     });
     await Promise.all(promises); // Wait for all browsers to finish
   }
 
+  // TODO: Finish this method with real data.
+  async _fetchMissingOneImplemenationCounts() {
```

Comment:
This purpose of this is to speed up the fetch right? e.g. find the nearest window that has the pre-calculated data

Reply:
# About the window arguments in the pre-calculation method
We only run the pre-calculation during the ingestion workflow to populate the tables for future queries. We do not calculate on the fly (i.e. if someone queries outside the data window specified, they will get nothing)


To prevent users from getting no data during their queries in production, we precalculate all the data with a much bigger window[[1](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/lib/gcpspanner/spanneradapters/web_features_consumer.go#L105-L110)][[2](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/workflows/steps/services/web_feature_consumer/cmd/job/main.go#L67-L74)][[3](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/infra/ingestion/workflows.tf#L57-L58)].

# Why even have those arguments

It takes a lot of space compared to other [tables](https://screenshot.googleplex.com/C2YhWf2dpZiMjGx). Space wise in production, it is fine. But locally, we need to limit how much data we actually ingest. Otherwise, CI can possibly run [out of disk space](https://github.com/GoogleChrome/webstatus.dev/pull/117). That's what that tuning knob is for.

# In this case
To get data for these counts on the stats page, we should:
1. Create a [new variable](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/util/cmd/load_fake_data/main.go#L407-L408) that starts 12 months before the existing mock end date of [dec 1, 2023](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/e2e/tests/stats-page.spec.ts#L21). (instead of using the current start window of Jan 1, 2020)

Reply:
Also, one other thing to note, it seems that the stats page test is not respecting the fake time like the features page test. The charts on the feature detail page are starting at their mock time of [dec 1, 2020](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/e2e/tests/utils.ts#L19).

![image](https://github.com/user-attachments/assets/ce37b946-31fd-4176-84de-76ac0b102af2)

I see now that the stats page test is not respecting the [Dec 1, 2023](https://github.com/GoogleChrome/webstatus.dev/blob/2901887e4c0201aa40c0a2304c24669c590f026a/e2e/tests/stats-page.spec.ts#L21C28-L21C49) mock time.

That may be another problem you will need to solve too.

Reply:
Thanks for the explanation James. The stats screenshots seem to align with the Dec 1, 2023 starting time
![Screenshot 2024-12-16 2 42 25 PM](https://github.com/user-attachments/assets/3f0f8a9c-abbc-49e4-a920-2eefe4d2b39e)

Regardless, I will send out a follow-up PR for the comments above!